### PR TITLE
Epochs Module

### PR DESCRIPTION
**Pull request by AustinoBombino**

### CODE REVIEW

**Comment by cosmic-vagabond** [comment]
looks good so far thanks for working on that.

File: proto/elys/epochs/v1/query.proto [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+
+option go_package = 	"github.com/elys-network/elys/x/epochs/types";
+
+// Query defines the gRPC querier service.
+service Query {
+  // EpochInfos provide running epochInfos
+  rpc EpochInfos(QueryEpochsInfoRequest) returns (QueryEpochsInfoResponse) {
+    option (google.api.http).get = "/evmos/epochs/v1/epochs";
```
Comment (cosmic-vagabond) [inline proto/elys/epochs/v1/query.proto:15]: @AustinoBombino can we update the API endpoints as well so it uses the same base endpoint as the other modules endpoints see here 
image

thanks

File: docs/static/openapi.yml [hunk 2]
```diff
@@ -471,6 +471,7 @@ paths:
             type: object
```
Comment (cosmic-vagabond) [inline docs/static/openapi.yml:471]: @AustinoBombino there are a lot of changes that seem to have happened in this file that will be very difficult to review, do you mind explain in providing a summary of what changed here thanks

File: x/epochs/spec/01_concepts.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<!--
+order: 1
+-->
+
+# Concepts
+
+The `epochs` module defines on-chain timers that execute at fixed time intervals. Other Evmos modules can then register logic to be executed at the timer ticks. We refer to the period in between two timer ticks as an "epoch".
```
Comment (cosmic-vagabond) [inline x/epochs/spec/01_concepts.md:7]: @AustinoBombino I like the EVMOS documentation structure I was thinking of using the same tools to generate the protocol and broader documentation of Elys based on those spec files.

here maybe a minor change to rename the occurrences of `Evmos` to `Elys` thanks

File: x/epochs/spec/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+title: "Epochs Overview"
+parent:
+  title: "epochs"
+-->
+
+# `epochs`
+
+## Abstract
+
+This document specifies the internal `x/epochs` module of the Evmos Hub.
```
Comment (cosmic-vagabond) [inline x/epochs/spec/README.md:12]: @AustinoBombino same here

File: docs/static/openapi.yml (hunk 2 above)
Comment (AustinoBombino) [reply docs/static/openapi.yml:471]: I think these are just changes that Ignite generated for the openapi docs

**Review by cosmic-vagabond — APPROVED** [review]
LGTM

File: docs/static/openapi.yml (hunk 2 above)
Comment (cosmic-vagabond) [reply docs/static/openapi.yml:471]: @AustinoBombino oh gotcha in that case we might need to make sure we are using the same version of ignite, here is mine: 

```
$ ignite version
Ignite CLI version:		v0.26.1
Ignite CLI build date:		2023-01-24T18:50:18Z
Ignite CLI source hash:		d2a2e287c176ba010557f0b8fbaa198cdeee928d
Ignite CLI config version:	v1
Cosmos SDK version:		v0.46.7
Your OS:			linux
Your arch:			amd64
Your Node.js version:		v19.3.0
Your go version:		go version go1.19.4 linux/amd64
[...]
```

Comment (AustinoBombino) [reply docs/static/openapi.yml:471]: @cosmic-vagabond 
```
$ignite version
Ignite CLI version:             v0.26.1
Ignite CLI build date:          2023-01-24T18:50:18Z
Ignite CLI source hash:         d2a2e287c176ba010557f0b8fbaa198cdeee928d
Ignite CLI config version:      v1
Cosmos SDK version:             v0.46.7
Your OS:                        linux
Your arch:                      amd64
Your Node.js version:           v18.9.0
Your go version:                go version go1.19 linux/amd64
```

**Pull request merged**